### PR TITLE
Update upwinding based on FCT

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR updates the upwinding to use CD2 in the implicit and a correction term (FCT-CD2) for the explicit tendency.

Co-authored by @dennisYatunin.

### CODE REVIEW

File: src/prognostic_equations/implicit/implicit_solver.jl [hunk 1]
```diff
@@ -357,7 +357,7 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ, colidx)
     (; ᶜspecific, ᶠu³, ᶜK, ᶜts, ᶜp, ᶜΦ, ᶠgradᵥ_ᶜΦ, ᶜρ_ref, ᶜp_ref) = p
     (; ∂ᶜK_∂ᶜuₕ, ∂ᶜK_∂ᶠu₃, ᶠp_grad_matrix, ᶜadvection_matrix) = p
     (; ᶜdiffusion_h_matrix, ᶜdiffusion_u_matrix, params) = p
-    (; energy_upwinding, density_upwinding) = p.atmos.numerics
+    (; energy_upwinding) = p.atmos.numerics
```
Comment (dennisYatunin) [inline src/prognostic_equations/implicit/implicit_solver.jl:360]: We also need to remove `energy_upwinding`, `tracer_upwinding`, and `precip_upwinding` from this function. They should all be set to `Val(:none)` here.

File: src/prognostic_equations/implicit/implicit_tendency.jl [hunk 2]
```diff
@@ -77,26 +119,15 @@ vertical_advection!(ᶜρχₜ, ᶠu³, ᶜχ, ::Val{:third_order}) =
     @. ᶜρχₜ -= ᶜadvdivᵥ(ᶠupwind3(ᶠu³, ᶜχ)) - ᶜχ * ᶜadvdivᵥ(ᶠu³)
 
 function implicit_vertical_advection_tendency!(Yₜ, Y, p, t, colidx)
-    (; energy_upwinding, density_upwinding) = p.atmos.numerics
-    (; tracer_upwinding, precip_upwinding) = p.atmos.numerics
```
Comment (dennisYatunin) [inline src/prognostic_equations/implicit/implicit_tendency.jl:81]: We should not be removing `precip_upwiding` from this tendency. Precipitation always needs to be upwinded in order for microphysics to work correctly.

File: src/prognostic_equations/advection.jl [hunk 3]
```diff
@@ -111,6 +115,72 @@ NVTX.@annotate function explicit_vertical_advection_tendency!(Yₜ, Y, p, t)
...
+                top = Operators.SetValue(C3(FT(0))),
+                # bottom = Operators.SetDivergence(FT(0)),
+            )
+
+            @. ᶠu³ₚ[colidx] =
+                FT(-1) *
+                ᶠinterp(p.precomputed.ᶜwₛ[colidx]) *
+                CT3(unit_basis_vector_data(CT3, lgf[colidx]))
+            @. ᶜqₚ[colidx] = Y.c.ρq_sno[colidx] / Y.c.ρ[colidx]
+        end
```
Comment (dennisYatunin) [inline src/prognostic_equations/advection.jl:180]:
```suggestion
```

**Comment by charleskawczynski** [comment]
I know we discussed this already, @szy21 and @dennisYatunin, but I wanted to ping you for last minute review (I've added this PR to the merge queue, so feel free to stop if something looks wrong).

**Review by dennisYatunin — APPROVED** [review]

**Review by szy21 — APPROVED** [review]

**Pull request merged**